src/Uno.Toolkit.UI/Controls/NavigationBar/NavigationBarRenderer.iOS.cs

```diff
@@ -77,58 +78,156 @@ protected override void Render()
+					}
 					break;
 				default: //Background is null
-					Native.BarTintColor = null;
-					Native.SetBackgroundImage(null, UIBarMetrics.Default); // Restores the default blurry background
-					Native.ShadowImage = null; // Restores the default 1px line
-					Native.Translucent = true;
+					if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+					{
+						appearance.ConfigureWithDefaultBackground();
+						appearance.BackgroundColor = null;
+					}
+					else
+					{
+						Native.BarTintColor = null;
+						Native.SetBackgroundImage(null, UIBarMetrics.Default); // Restores the default blurry background
+						Native.ShadowImage = null; // Restores the default 1px line
+						Native.Translucent = true;
+					}
 					break;
 			}
 
+			// Foreground
+			if (ColorHelper.TryGetColorWithOpacity(Element.Foreground, out var foregroundColor))
+			{
+				if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+				{
+					appearance.TitleTextAttributes = new UIStringAttributes
+					{
+						ForegroundColor = foregroundColor,
+					};
+
+					appearance.LargeTitleTextAttributes = new UIStringAttributes
+					{
+						ForegroundColor = foregroundColor,
+					};
+				}
+				else
+				{
+					Native.TitleTextAttributes = new UIStringAttributes
+					{
+						ForegroundColor = foregroundColor,
+					};
+				}
+			}
+			else
+			{
+				if(UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+				{
+					appearance.TitleTextAttributes = new UIStringAttributes();
+					appearance.LargeTitleTextAttributes = new UIStringAttributes();
+				}
+				else
+				{
+					Native.TitleTextAttributes = null;
+				}
+			}
+
 			var mainCommand = Element.GetValue(NavigationBar.MainCommandProperty) as AppBarButton;
 
-			// CommandBarExtensions.BackButtonForeground
-			ColorHelper.TryGetColorWithOpacity(mainCommand?.Foreground, out var backButtonForeground);
-			Native.TintColor = backButtonForeground;
+			// MainCommand.Foreground
+			ColorHelper.TryGetColorWithOpacity(mainCommand?.Foreground, out var mainCommandForeground);
+			Native.TintColor = mainCommandForeground;
+
+			// MainCommand.Icon
+			var mainCommandIcon = mainCommand?.Icon is BitmapIcon bitmapIcon
+				? ImageHelper.FromUri(bitmapIcon.UriSource)?.ImageWithRenderingMode(UIImageRenderingMode.AlwaysOriginal)
+				: null;
+
+			if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+			{
+				var backButtonAppearance = new UIBarButtonItemAppearance(UIBarButtonItemStyle.Plain);
+
+				if (mainCommandForeground is { } foreground)
+				{
+					var titleTextAttributes = new UIStringAttributes
+					{
+						ForegroundColor = foreground
+					};
+
+					var attributes = new NSDictionary<NSString, NSObject>(
+						new NSString[] { titleTextAttributes.Dictionary.Keys[0] as NSString }, titleTextAttributes.Dictionary.Values
+					);
+
+					backButtonAppearance.Normal.TitleTextAttributes = attributes;
+					backButtonAppearance.Highlighted.TitleTextAttributes = attributes;
+
+					if (mainCommandIcon is { } image)
```

Comment:
so we dont set the icon if there is no foreground color?

Reply:
wooops